plugins/jobs/src/js/data/JobModel.ts

```diff
@@ -0,0 +1,293 @@
+import { makeExecutableSchema, IResolvers } from "graphql-tools";
+import { Observable } from "rxjs";
+
+export interface Query {
+  jobs: JobConnection | null;
+  job: Job | null;
+}
+
+export interface GeneralArgs {
+  [key: string]: any;
+}
+
+export interface JobsQueryArgs {
+  filter?: string | null;
+  sortBy?: SortOption | null;
+  sortDirection?: SortDirection | null;
+}
+
+export interface JobQueryArgs {
+  id: string;
+  filter?: string | null;
+  sortBy?: SortOption | null;
+  sortDirection?: SortDirection | null;
+}
+
+export type SortOption = "ID" | "STATUS" | "LAST_RUN";
+
+export type SortDirection = "ASC" | "DESC";
+
+export interface JobConnection {
+  filteredCount: number;
+  totalCount: number;
+  nodes: Job[];
+}
+
+export interface Job {
+  activeRuns: JobRunConnection;
+  command: string;
+  cpus: number;
+  description: string;
+  disk: number;
+  docker: Docker | null;
+  id: string;
+  jobRuns: JobRunConnection;
+  json: string;
+  labels: Label[];
+  lastRunsSummary: JobHistorySummary;
+  lastRunStatus: JobRunStatusSummary;
+  mem: number;
+  name: string;
+  schedules: ScheduleConnection;
+  scheduleStatus: JobStatus;
+}
+
+export interface JobRunConnection {
+  longestRunningActiveRun: JobRun | null;
+  nodes: JobRun[];
+}
+
+export interface JobRun {
+  dateCreated: number;
+  dateFinished: number;
+  jobID: string;
+  status: JobRunStatusSummary;
+  tasks: JobTaskConnection;
+}
+
+export interface JobRunStatusSummary {
+  status: JobRunStatus | null;
+  time: number | null;
+}
+
+export type JobRunStatus = "FAILED" | "NOT_AVAILABLE" | "SUCCESS";
+
+export interface JobTaskConnection {
+  longestRunningTask: JobTask;
+  nodes: JobTask[];
+}
+
+export interface JobTask {
+  dateCompleted: number;
+  dateStarted: number;
+  status: JobTaskStatus;
+  taskId: string;
+}
+
+export type JobTaskStatus =
+  | "TASK_CREATED"
+  | "TASK_DROPPED"
+  | "TASK_ERROR"
+  | "TASK_FAILED"
+  | "TASK_FINISHED"
+  | "TASK_GONE"
+  | "TASK_GONE_BY_OPERATOR"
+  | "TASK_KILLED"
+  | "TASK_KILLING"
+  | "TASK_LOST"
+  | "TASK_RUNNING"
+  | "TASK_STAGING"
+  | "TASK_STARTED"
+  | "TASK_STARTING"
+  | "TASK_UNKNOWN"
+  | "TASK_UNREACHABLE";
+
+export interface Docker {
+  forcePullImage: boolean;
+  image: string;
+}
+
+export interface Label {
+  key: string;
+  value: string;
+}
+
+export interface JobHistorySummary {
+  failureCount: number;
+  lastFailureAt: string | null;
+  lastSuccessAt: string | null;
+  successCount: number;
+}
+
+export interface ScheduleConnection {
```

Comment:
How do you get away not putting "I" on all interfaces here. I think this is a good example on why the pattern to start interfaces with the letter "I" should die.

Reply:
Disabled it in the tslint config. We discussed it in the sync and we moved the item a couple of times by saying we still need to find some strong cases for it. Had a short discussion with @orlandohohmeier on that topic and we didn't want to introduce so many types that we would need to rename later on, possibly causing confusion.